bundles/org.eclipse.core.databinding.property/src/org/eclipse/core/internal/databinding/property/value/MapDelegatingValueObservableMap.java

```diff
@@ -162,36 +162,34 @@ private MapDiff<K, V> convertDiff(MapDiff<? extends K, ? extends I> diff) {
 			Map<K, V> newValues = new HashMap<>();
 
 			Set<? extends K> addedKeys = diff.getAddedKeys();
-			for (K key : addedKeys) {
+			addedKeys.forEach(key -> {
 				I masterValue = diff.getNewValue(key);
 				V newValue = cache.get(masterValue);
 				newValues.put(key, newValue);
-			}
+			});
 
 			Set<? extends K> removedKeys = diff.getRemovedKeys();
-			for (K key : removedKeys) {
+			removedKeys.forEach(key -> {
 				I masterValue = diff.getOldValue(key);
 				V oldValue = cache.get(masterValue);
 				oldValues.put(key, oldValue);
-			}
+			});
 
 			Set<K> changedKeys = new HashSet<>(diff.getChangedKeys());
-			for (Iterator<K> it = changedKeys.iterator(); it.hasNext();) {
-				K key = it.next();
-
+			changedKeys.forEach(key -> {
 				I oldMasterValue = diff.getOldValue(key);
 				I newMasterValue = diff.getNewValue(key);
 
 				V oldValue = cache.get(oldMasterValue);
 				V newValue = cache.get(newMasterValue);
 
 				if (Objects.equals(oldValue, newValue)) {
-					it.remove();
+					changedKeys.remove(key);
```

Comment:
I think iterator is required here because otherwise removing items from set might throw concurrent modification exception or is it guaranteed?

Reply:
Thanks for the suggestion, made the changes by making use of `removeIf` API.